### ISSUE TITLE
Add option to create an overlay image/layer that can be used for reference

### ISSUE THREAD

**Issue by dacap** [aseprite/aseprite#532]
This overlay image could be bigger than the current sprite size, so the zoom wouldn't affect it and we could see our small sprite with an overlay image of huge resolution as background.

Reference: http://www.aseprite.org/faq/#comment-1694009857
![original](https://cloud.githubusercontent.com/assets/39654/9143779/b9d603fc-3d1e-11e5-839e-b9a1806efb43.jpg)

**Comment by vaultec** [comment]
Is this in the program? This is a feature I've long been waiting to be implemented. This would help so much in drawing pixel art based on a sketch.

**Comment by dacap** [comment]
Hi @vaultec, this feature isn't yet in the program. I'm not sure if this will be included in v1.1 or v1.2, but it will be available in a near future.

**Comment by WingedRayeth** [comment]
Awesome to hear it will be available soon. I have been waiting on this since I suggested it!

**Comment by AlbertoFdzM** [comment]
+1 :+1:

**Comment by Ravetracer** [comment]
+1 
The mobile editor "Pixly" already has this feature and it's a big help!

**Comment by dacap** [comment]
As it's one of the most requested features, a simple version of this feature might appear in v1.1.

**Comment by jameswilddev** [comment]
I'd like to see this too.  You can currently import on a bottom layer, but that means you can't have full colour on this bottom layer and then indexed colour above.

**Comment by robertocaldas** [comment]
Is this happening? :)

**Comment by dacap** [comment]
@robertocaldas I'd love to have this as soon as possible, maybe next week? :eyes:

**Comment by robertocaldas** [comment]
So awesome!

Is it a totally lame idea to support gif, being able to select the gif frame as a reference image?
I'm totally talentless, my only hope is to be "heavily inspired" by other people's work, if you know what I mean...  *sigh*

**Comment by dacap** [comment]
This feature needs a lot of internal work to be done correctly (e.g. make any layer a reference layer so we can support animation references). The work needed for this is quite related to this feature #593 and it might be a good idea to prepare everything for #454 and #459 

I think this will be the first feature opening v1.2 which I'd like to make it available on Steam beta branch. And as a first step, I'll make it something really simple: Select another file as reference from the Sprite > Properties. If I have enough time, I'll try to release the official release with fixes, and the beta branch on Steam.

**Comment by robertocaldas** [comment]
Sorry to be pushy, but do you have an ETA on the first version?

**Comment by Enjineer30302** [comment]
Yeah, I agree, because right now, it's been difficult to do sprites from some concept art. Being able to draw over the image would be awesome! :D

**Comment by Saphiresurf** [comment]
Hey, just so you guys know I'd also be someone who would take use of this! It'd be absolutely lovely because it gives me an easy way to recreate styles I really love in a more pixel type formatting. Just wonderful for practice and learning new things in general. Hope to see this come to fruition soon!

**Comment by Saphiresurf** [comment]
Also, do you know the progress on this by any chance if there's been any?

**Comment by dacap** [comment]
As in [one of my previous comments](https://github.com/aseprite/aseprite/issues/532#issuecomment-215458721) this will take some time. Mainly because it needs some special UI to move/scale the overlay in the desired location and a new render code. (Back in those days I thought it would be easy enough to be implemented quickly, but I was wrong.)

**Comment by mattjb86** [comment]
Any update on this?  It's definitely the one feature I would most like to see implemented as I have a ton of reference sketches and having it would increase my workflow enormously.

**Comment by dacap** [comment]
I'll start working on this right now 😮

**Comment by bantonioch** [comment]
mattjb86 i don't know. but david added the "high priority" tag about a week ago, so maybe he would start working on this from now.

**Comment by bantonioch** [comment]
wow.... weird. commented at the same time. sorry, dacap. i guess i was right , then

**Comment by dacap** [comment]
Pretty close, coming soon on v1.2-beta4:

![finegrain](https://cloud.githubusercontent.com/assets/39654/19291597/87b4e8a8-8fec-11e6-8ec6-3bc3a8d2e92f.gif)

**Comment by bantonioch** [comment]
moooy bueeno! .. I mean.. Looks very promising!

**Comment by mattjb86** [comment]
Looking good!  Thanks for the work you've been putting into it, much appreciated!

**Comment by dacap** [comment]
![ref-layer-upload](https://cloud.githubusercontent.com/assets/39654/19371419/3137fb98-9189-11e6-827c-a40d4588fb72.gif)

**Comment by Saphiresurf** [comment]
Oh man this is looking awesome :)

**Comment by Ravetracer** [comment]
Looks really awesome. This is a new killer feature for Aseprite!

**Comment by bantonioch** [comment]
Hubo algún progreso al día de hoy?

**Comment by dacap** [comment]
@bantonioch hola, actualmente aseprite está en [v1.2-beta3](https://www.aseprite.org/release-notes/#aseprite-v1-2-beta3), esta nueva característica aparecerá en v1.2-beta4.

@bantonioch hi, actually aseprite is in [v1.2-beta3](https://www.aseprite.org/release-notes/#aseprite-v1-2-beta3), this new feature will appear in v1.2-beta4.

**Comment by dacap** [comment]
There is one case about this issue that worries me most: An indexed sprite with a RGB reference layer. The actual code converts the RGB reference to indexed, but this is not what a regular user would expect. The program should show the RGB reference image as it were RGB (without taking care about the indexed palette). This case needs that a sprite may contain different color modes on each layer (e.g. an indexed layer, and a rgb layer).

I'll see that will be available as a first version of this feature to avoid introduced bugs or unexpected behaviors.

**Comment by dacap** [comment]
Will be available in next release (I hope this week). Merged impl in bc1ad97fb5eb927c4e6788ddaf8a0759b11a2aae

Future work in #1321

**Comment by grymmjack** [comment]
FANTASTIC! Thank you so much @dacap this is a really welcome feature addition!

**Comment by Saphiresurf** [comment]
Seriously it's really appreciated! Going to come in handy for rotoscoping and such I'm actually really excited to start working on stuff with this.

**Comment by charyoshi** [comment]
Maybe I'm missing something here but how exactly is this done? I can't manage to load an image at one resolution and draw on it with another, and the closest thing to an answer I've found so far is "it really works!"

**Comment by dacap** [comment]
Hi @charyoshi, first of all you need the [beta version](https://www.aseprite.org/beta/). After that you will be able to use the Layer > Add Reference Layer menu option to select your own .png or .gif image as a background reference layer.

**Comment by charyoshi** [comment]
@dacap Thank you so much ;-;

**Comment by ibrahimokur** [comment]
Thank you
